Source/monster.cpp

```diff
@@ -50,6 +50,19 @@ void NewMonsterAnim(int i, AnimStruct *anim, Direction md, AnimationDistribution
 	monst->_mdir = md;
 }
 
+void StartMonsterGotHit(int mid)
+{
+	if (Monsters[mid].MType->mtype != MT_GOLEM) {
+		NewMonsterAnim(mid, &Monsters[mid].MType->Anims[MA_GOTHIT], Monsters[mid]._mdir, gGameLogicStep < GameLogicStep::ProcessMonsters ? AnimationDistributionFlags::ProcessAnimationPending : AnimationDistributionFlags::None);
+		Monsters[mid]._mmode = MM_GOTHIT;
+	}
+	Monsters[mid].position.offset = { 0, 0 };
+	Monsters[mid].position.tile = Monsters[mid].position.old;
+	Monsters[mid].position.future = Monsters[mid].position.old;

```

Comment:
I wonder if you considered abstracting this "sequence" of actions here as a member function on `position` itself. Perhaps something like `stop()` or similar.

EDIT: Or as a member function on the `MonstStruct` itself even (might make even more sense).

Reply:
I found the same sequence at two other places (death and diablo death).
But I think this can be done later and not part of this pr.